src/common.h

```diff
@@ -146,11 +146,24 @@ void __inline CLIB_DECL D_(const char *text, ...) { do {} while (0); }
 #define unlink _unlink
 #endif
 #if defined(_WIN32) || defined(__WATCOMC__) /* in win32.c */
+/* MSVC 2015+ has C99 compliant snprintf and vsnprintf implementations.
+ * If __USE_MINGW_ANSI_STDIO is defined for MinGW (which it is by default),
+ * compliant implementations will be used instead of the broken MSVCRT
+ * functions. Additionally, GCC may optimize some calls to those functions. */
+#ifdef _MSC_VER
+#if _MSC_VER < 1900
+#define USE_LIBXMP_SNPRINTF
+#endif
+#elif defined(__WATCOMC__) || !defined(__USE_MINGW_ANSI_STDIO)
```

Comment:
Nice catch, I forgot it could explicitly be defined to 0 and didn't find much documentation on it while looking into it. Working on applying changes from your patch right now.

Reply:
See full patch suggestion at https://github.com/cmatsuoka/libxmp/pull/175#issuecomment-703331109

Reply:
I applied the relevant parts of your patch (cleaning up the macro logic, __USE_MINGW_ANSI_STDIO=0, and the win32.c fix).